group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -10117,8 +10130,9 @@ public void testConsumerGroupHeartbeatWithPreparingRebalanceClassicGroup() throw
         assertEquals(group, context.groupMetadataManager.getOrMaybeCreateClassicGroup("group-id", false));
     }
 
-    @Test
-    public void testConsumerGroupHeartbeatToClassicGroupFromExistingStaticMember() {
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testConsumerGroupHeartbeatToClassicGroupFromExistingStaticMember(boolean isConsumerGeneratedMemberId) {
```

Comment:
Why are we changing those tests in particular?

Reply:
Originally, I was thinking covering both scenarios for client-generated memberId and server-generated memberId. However, based on the previous comment, I will focus on simulating the behavior of the latest version and update the test case accordingly.